src/test/scala/org/broadinstitute/dsde/firecloud/service/WorkspaceApiServiceSpec.scala

```diff
@@ -848,6 +825,44 @@ class WorkspaceApiServiceSpec extends BaseServiceSpec with WorkspaceApiService {
       }
     }
 
+    "Workspace updateAttributes tests" - {
+      "when calling any method other than PATCH on workspaces/*/*/updateAttributes path" - {
+        "should receive a MethodNotAllowed error" in {
+          List(HttpMethods.PUT, HttpMethods.POST, HttpMethods.GET, HttpMethods.DELETE) map {
+            method =>
+              new RequestBuilder(method)(updateAttributesPath, HttpEntity(MediaTypes.`application/json`, "{}")) ~> dummyUserIdHeaders("1234") ~> sealRoute(workspaceRoutes) ~> check {
+                status should equal(MethodNotAllowed)
+              }
+          }
+        }
+      }
+
+      "when calling PATCH on workspaces/*/*/updateAttributes path" - {
+        "should 400 Bad Request if the payload is malformed" in {
+          (Patch(updateAttributesPath, HttpEntity(MediaTypes.`application/json`, "{{{"))
+            ~> dummyUserIdHeaders("1234")
+            ~> sealRoute(workspaceRoutes)) ~> check {
+            status should equal(BadRequest)
+          }
+        }
+
+        "should 200 OK if the payload is ok" in {
+          (Patch(updateAttributesPath,
+            HttpEntity(MediaTypes.`application/json`, """[
+                                                        |  {
+                                                        |    "op": "AddUpdateAttribute",
+                                                        |    "attributeName": "library:dataCategory",
+                                                        |    "addUpdateAttribute": "test-attribute-value"
+                                                        |  }
+                                                        |]""".stripMargin))
+            ~> dummyUserIdHeaders("1234")
+            ~> sealRoute(workspaceRoutes)) ~> check {
+            status should equal(OK)
+          }
+        }
+      }
+    }
+
```

Comment:
I'd love to see some tests that actually check publishing when calling the attribute endpoints!